### DISCUSSION TITLE
Access return value of functions

### DISCUSSION THREAD

**Discussion by DeCabMax** [zauberzeug/nicegui#1831] [Q&A] [answered]
### Question

Hi,
i want to read the the return variable of a function that is triggerd by ui.button.
How do i access the returned variable. z.value doesnt work because z is a button and have no value. 
```python
def function(x,y):
    return_v1 = np.sqrt(x**2 + y**2 )
    return return_v1

x =  ui.number(label='x',format='%.4f', value = 0)
y =  ui.number(label='x',format='%.4f', value = 0)

z = ui.button(on_click= lambda e:function_1(x.value,y.value))

ui.label(      )
##          A
##          |
## here i want to show the return value 

```

**Comment by rodja** [comment, marked as answer]
Straying a bit further away from your code, I would do it with bindings:

```py
@dataclass
class Coordinate:
    x: float
    y: float

    @property
    def z(self) -> float:
        return np.sqrt((self.x or 0)**2 + (self.y or 0)**2)

c = Coordinate(0, 0)
ui.number(label='x', format='%.4f').bind_value(c, 'x')
ui.number(label='y', format='%.4f').bind_value(c, 'y')
ui.label().bind_text_from(c, 'z', backward=lambda z: f'z={z:.4f}')
```

**Comment by falkoschindler** [comment]
Regarding the original question: Note that when `ui.label` is placed on the page, the button hasn't been clicked and the function hasn't been executed yet. Therefore there is no return value at this moment.

To update the label, you can use binding like in @rodja's answer. Or you let the event handler update the label:
```py
def function_1(x: float, y: float):
    z.text = f'{np.sqrt(x**2 + y**2):.4f}'

x = ui.number(label='x', format='%.4f', value=0)
y = ui.number(label='y', format='%.4f', value=0)
ui.button('Compute', on_click=lambda: function_1(x.value, y.value))

z = ui.label()
```

**Reply by rohitsathish** [reply]
How would this approach be applied for tables @falkoschindler? As I understand, you can't have an empty instantiation of a table.

**Reply by falkoschindler** [reply]
I think this is being discusses over at https://github.com/zauberzeug/nicegui/discussions/2098.
